docs/rule/require-each-key.md

```diff
@@ -0,0 +1,82 @@
+# require-each-key
+
+In order to improve rendering speed, Ember will try to reuse the DOM elements where possible. Specifically, if the same item is present in the array both before and after the change, its DOM output will be reused.
+
+The key option is used to tell Ember how to determine if the items in the array being iterated over with {{#each}} has changed between renders. By default the item's object identity is used.
+
+This is usually sufficient, so in most cases, the key option is simply not needed. However, in some rare cases, the objects' identities may change even though they represent the same underlying data.
```

Comment:
I don't have the actual stats, but it seems to me that "in most cases" and "in some rare cases" are exaggerations. Managing data in an immutable fashion is trending, many frameworks/state management libraries require immutability, and even Glimmer docs recommend using "The Immutable Pattern". The current wording might make the problem look small and irrelevant. But for us, the problem was extensive and severe.